### PR TITLE
Fixes #34653 - Rename SSH provider to script

### PR DESCRIPTION
**Pull request by adamruzicka**
Currently the new provider is just an alias to the SSH provider to make
the transition easier and not require making this change across all
dependenat plugins at once.

### CODE REVIEW

File: app/models/ssh_execution_provider.rb [hunk 1]
```diff
@@ -64,3 +64,6 @@ def ssh_port(host)
     end
   end
 end
+
+class ScriptExecutionProvider < SSHExecutionProvider
```
Comment (ezr-ondrej) [inline app/models/ssh_execution_provider.rb:68]: don't we want to alias that instead and ideally rename to `ScriptExecutionProvider` and make the SSHExecutionProvider an alias?
```suggestion
SSHExecutionProvider = ScriptExecutionProvider
```
if there are concerns for doing it in this PR, then I'd do it the other way arround, but not subclassing, unless we really want to subclass for some reason.

**Review by ezr-ondrej — CHANGES_REQUESTED** [review]
This fails for me with `Failed to initialize: RuntimeError - Could not use any proxy for the script job. Consider configuring remote_execution_global_proxy, remote_execution_fallback_proxy in settings` I believe we need to register `script` as an ability for proxy first?

**Comment by adamruzicka** [comment]
But then even the proxy would have to report the feature as script, which I'm not sure we want. I definitely don't like that idea. How about letting a provider define a list of proxy feature aliases that would be used when picking a proxy?

**Comment by ezr-ondrej** [comment]
can't we have proxy report it as both `ssh` and `script` aliasing the same way we do here?

File: app/models/remote_execution_provider.rb [hunk 2]
```diff
@@ -4,7 +4,15 @@ class RemoteExecutionProvider
 
   class << self
     def provider_for(type)
-      providers[type.to_s] || providers[:SSH]
+      providers[type.to_s] || providers[:script]
+    end
+
+    def registered_name
+      providers.find { |_name, klass| klass == self }.first
```
Comment (ezr-ondrej) [inline app/models/remote_execution_provider.rb:11]: This is bit hard to ready, could we use https://ruby-doc.org/core-2.7.0/Hash.html#method-i-key ?
```suggestion
      klass = self
      providers.key(klass)
```

**Review by ezr-ondrej — APPROVED** [review]
Works well now :+1:
I'd imagine we want to rename the proxy feature as well, but now we have the option to do it decoupled from this PR :+1:

File: app/models/remote_execution_provider.rb (hunk 2 above)
Comment (adamruzicka) [reply app/models/remote_execution_provider.rb:11]: TIL

**Comment by ezr-ondrej** [comment]
I'd like to keep both commits separate as it make sense IMHO, would it be an issue to have both with fixes keyword?

**Pull request merged**